tests/test_multi_mapping.py

```diff
@@ -0,0 +1,261 @@
+    )
+
+    assert 0 == result.exit_code
+
+    [s2t, t2s] = csvrow.back_get(output / "person_ids.tsv")
+    assert 2 == len(s2t)
+    assert 2 == len(t2s)
+
+    # check the birthdateime and gender
+    seen_ids: list[int] = []
+    for person in csvrow.csv_rows(output / "person.tsv", "\t"):
+        # check the ids
+        assert person.person_id in t2s
+        src_person_id = int(t2s[person.person_id])
+        assert src_person_id not in seen_ids
+        seen_ids.append(src_person_id)
+
+        # dataset is small so just check it like this
+        if 28 == src_person_id:
+            assert "8532" == person.gender_concept_id
+            assert "2020-09-12 00:00:00" == person.birth_datetime
+        elif 82 == src_person_id:
+            assert "8507" == person.gender_concept_id
+            assert "2020-09-11 00:00:00" == person.birth_datetime
+        else:
+            raise Exception(f"unexpected {src_person_id=} in {person=}")
+    assert 28 in seen_ids
+    assert 82 in seen_ids
+
+    # ethnicity is an observation
+    for observation in csvrow.csv_rows(output / "observation.tsv", "\t"):
+        assert observation.person_id in t2s
+        src_person_id = int(t2s[observation.person_id])
+        if 28 == src_person_id:
+            assert "40618782" == observation.observation_concept_id
+            assert "2020-09-12" == observation.observation_date
+        elif 82 == src_person_id:
+            assert "40637268" == observation.observation_concept_id
+            assert "2020-09-11" == observation.observation_date
+        else:
+            raise Exception(f"unexpected {src_person_id=} in {observation=}")
+
+
+@pytest.mark.unit
+def test_observe_smoking(tmp_path: Path):
+    arg__input_dir = Path(__file__).parent / "test_multi_mapping/test_observe_smoking"
+    arg__rules_file = arg__input_dir / "mapping.json"
+
+    ##
+    #
+    (result, output) = clicktools.click_mapstream(
+        tmp_path,
+        ["demos.csv", "smoke.csv"],
+        arg__input_dir,
+        arg__rules_file,
+    )
+
+    assert 0 == result.exit_code
+
+    [s2t, t2s] = csvrow.back_get(output / "person_ids.tsv")
+    assert 4 == len(s2t)
+    assert 4 == len(t2s)
+
+    # declare the expectations
+    # ... it's a lot nicer to declare it like this
+    expect = {
+        123: {
+            "2018-01-01": "active",
+            "2018-02-01": "active",
+            "2018-03-01": "quit",
+            "2018-04-01": "active",
+            "2018-05-01": "never",
+        },
+        456: {
+            "2009-01-01": "never",
+            "2009-02-01": "never",
+            "2009-03-01": "quit",
+        },
+    }
+
+    concept = {
+        "active": "3959110",
+        "quit": "3957361",
+        "never": "35821355",
+    }
+
+    # now rewrite the expectation to use concept ids
+    for p in expect:
+        for d in expect[p]:
+            expect[p][d] = concept[expect[p][d]]
+
+    # check the smoking state in observations
+    seen: int = 0

```

Comment:
Could you rename `seen` to something more descriptive and explicit? Looks like this is repeated as well